src/coreclr/src/jit/importer.cpp

```diff
@@ -4226,6 +4226,46 @@ GenTree* Compiler::impMathIntrinsic(CORINFO_METHOD_HANDLE method,
 
     op1 = nullptr;
 
+    if (intrinsicID == CORINFO_INTRINSIC_Pow && impStackTop().val->IsCnsFltOrDbl())
+    {
+        double    power = impStackTop().val->AsDblCon()->gtDconVal;
+        var_types type  = impStackTop().val->TypeGet();
+        if (power == 2.0)
+        {
+            // Math.Pow(x, 2) -> x*x
+            impPopStack();
+            GenTree* arg0 = impPopStack().val;
+            if (arg0->OperIs(GT_LCL_VAR, GT_LCL_FLD, GT_CNS_DBL))
+            {
+                // we don't need to introduce a temp variable for the
+                // first Math.Pow argument if it's a lcl or cns
+                return gtNewOperNode(GT_MUL, type, arg0, gtCloneExpr(arg0));
+            }
+            unsigned temp = lvaGrabTemp(true DEBUGARG("pow arg"));
+            impAssignTempGen(temp, arg0, static_cast<unsigned>(CHECK_SPILL_NONE));
+            return gtNewOperNode(GT_MUL, type, gtNewLclvNode(temp, type), gtNewLclvNode(temp, type));
+        }
+        else if (power == 1.0)
+        {
+            // Math.Pow(x, 1) -> x
+            impPopStack();
+            return impPopStack().val;
+        }
+        else if (power == -1.0)
+        {
+            // Math.Pow(x, -1) -> 1/x
+            impPopStack();
+            return gtNewOperNode(GT_DIV, type, gtNewDconNode(1, type), impPopStack().val);
+        }
+        else if (power == 0.0)
+        {
+            // Math.Pow(x, 0) -> 1
+            impPopStack();
+            impPopStack();
+            return gtNewDconNode(1, type);
+        }
```

Comment:
oh, I guess I can't just throw `x` here away without checking for side-effects...